src/natalie_parser/parser.cpp

```diff
@@ -1193,6 +1197,41 @@ Node *Parser::parse_assignment_expression(Node *left, LocalsHashmap &locals) {
     }
 };
 
+Node *Parser::parse_assignment_expression(bool to_array, LocalsHashmap &locals, bool allow_multiple) {
```

Comment:
Let's rename this something like `parse_assignment_expression_value`. If I'm reading this right, it's just giving you back the right hand side of an assignment...